src/type/schema.js

```diff
@@ -232,7 +246,24 @@ export class GraphQLSchema {
     if (isUnionType(abstractType)) {
       return abstractType.getTypes();
     }
-    return this._implementations[abstractType.name] || [];
+
+    const types = this._objectImplementations[abstractType.name];
+    if (!types) {
+      return (this._objectImplementations[abstractType.name] = []);
+    }
```

Comment:
What is the motivation for this change?
It's increasing the size `_objectImplementations` and also allocates memory during execution. If you feel like `|| []` is a problem (if so can you please explain why?) you can put an empty array inside the constructor to have predictable performance less chance to hit performance cliff due to V8 deoptimization.

Reply:
While this does add some trivial allocation of empty arrays, it prevents a new array from being created every time `getPossibleTypes` is called

Reply:
I'm certainly willing to move this to the constructor, but I'm not convinced that the resulting behavior would be an improvement. If this were a realtime language like rust or C, I would understand a preference for time-predictable calls here over total allocation, and then we could avoid memoization in general. However, given that JS is garbage collected and a memoization pattern was already employed here, I think this is the "minimal change" that actually provides correct behavior.

Reply:
Perhaps an alternative which would reduce allocations while still preservice object identity for comparison purposes would be to simply hoist the empty array out of the function. I'll implement that instead.

Reply:
I've changed my mind here RE the importance of object identity. While it's a goal that I appreciate, I don't want to force on a project where it's not a shared by everyone involved, as it would be too easy for a consuming app to rely on object identity for the sake of comparisons, then break on a minor/patch update if that isn't considered part of the API. I'm going to take your recommendation here and just use `|| []`.

Reply:
I also changed my mind since interfaces and unions without subtypes should be pretty rare I moved this code into the constructor:
https://github.com/APIs-guru/graphql-js/blob/102a8d9eda29b6be52f37f4c17c857ae810895bb/src/type/schema.js#L335-L337